lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -1210,6 +1244,14 @@ LogicalResult OrOp::canonicalize(OrOp op, PatternRewriter &rewriter) {
   auto size = inputs.size();
   assert(size > 1 && "expected 2 or more operands");
 
+  // Skip if the operation should be flattened by another operation.
+  if (shouldBeFlattened(op))
+    return failure();
```

Comment:
Why short circuit all the other canonicalizations if you can't flatten?

Reply:
oh man i missed that. i think with your other suggestion to move this check into tryFlatteningOperands, we can have the previous behavior: if tryFlatteningOperands did flatten operands, success is returned, and otherwise the next canonicalizations are tried.

Reply:
The intention was to skip other canonicalizations flattening can be applied to the operation later to save compile times. I thought it would be much efficient to apply all canonicalizations after operands flattened but seems like there is not much difference in compile time practically. I moved the condition in the tryFlatteningOperands. Thank you for the suggestion.